### PR TITLE
React Table upgrade part 1

### PR DESCRIPTION
**Pull request by dgading**
This PR aims to recreate the core functionality of the old data tables with the new library version. The library APIs have changed a decent amount so it will probably take another sprint to finish cleaning any bugs and adding little touches like loading states and stuff. 

This should unblock at least the dataset page and the next ticket will focus on unblocking the loading of the filtered resource page.

### CODE REVIEW

**Review by brdunfield — COMMENTED** [review]
I'm not sure how many bugs you want to address in this PR vs logging them in other tickets.

I did notice that when you go to adjust the width of a column, the column snaps to the max width of the content within and can't be compressed past that. Adjusting a column width also appears to adjust column widths to the right of it

https://user-images.githubusercontent.com/5032149/223805596-d6deffd5-72fd-48a5-83b0-fe57df335a51.mov

File: src/components/Datatable/Datatable.jsx [hunk 1]
```diff
@@ -1,39 +1,198 @@
...
+    onSortingChange: setSorting,
+
+    getCoreRowModel: getCoreRowModel(),
+    getSortedRowModel: getSortedRowModel(),
+    debugTable: true,
+  });
+
+   React.useEffect(() => {
+    const normalizedSort = sortTransform ? sortTransform(sorting) : filters;
+    console.log(sorting)
```
Comment (brdunfield) [inline src/components/Datatable/Datatable.jsx:74]: A couple stray console logs left behind

**Comment by dgading** [comment]
There are two modes for the resize. I may try playing with the other one and see if it responds better. I definitely saw some issues when inspector was open because it is resizing all the stuff on the fly, so more columns means more overhead. There is an onEnd method that snaps it into place after the mouse button is released. I can tackle that in the next ticket, which dataset were you using for testing?

**Comment by brdunfield** [comment]
I was originally just testing on medicaid but switched over to PFS to make sure I'm just checking your work and not medicaid stuff that's not done.

They all seem to behave similarly, but here's one I'm using on my local: `/dataset/44f63ce9-430f-4d56-ba46-cfe26452862b`

**Review by brdunfield — COMMENTED** [review]
Another weird bug I notice is when dragging the column widths around, if the mouseup event happens to the left of the column border, it triggers a click event on the left cell. This doesn't happen on the right cell. 🤔

**Review by brdunfield — APPROVED** [review]
As long as we make tickets for these bugs, overall this LGTM!

**Comment by dgading** [comment]
Nice. In the current setup we have a maximum to try and limit the amount of x scrolling and resizing. In the redesigns we should be getting the full 1200+ px width of the page which should help some. I've always had issues with the resizing and the size of our tables, any feedback and testing notes will be great. I'll start a ticket for it for one of the next sprints so we can maybe work with UX to find the best solution.

**Pull request merged**